### ISSUE TITLE
Context menu on wrong position in Windows, crashes in Android

### ISSUE THREAD

**Issue by pekspro** [dotnet/maui#6155]
### Description

I have a ListView with ViewCell as base in the DataTemplate. The ViewCell has ContextActions. Triggering the context menu on Android throws an `System.NullReferenceException`. In Windows the menu is shown, but not where mouse pointer were when the user right clicked.

[Demo project](https://github.com/pekspro/MauiIssues/tree/BadContextMenuPosition)

### Steps to Reproduce

1. Create a new MAUI project.
2. Add this in the main page:
``` XAML
    <ListView x:Name="ItemsList">
        <ListView.ItemTemplate>
            <DataTemplate>
                <ViewCell>
                    <ViewCell.ContextActions>
                        <MenuItem Text="Detail"></MenuItem>
                        <MenuItem Text="Delete"></MenuItem>
                    </ViewCell.ContextActions>
                    <Grid>
                        <Label Text="{Binding }" FontSize="25"></Label>
                    </Grid>
                </ViewCell>
            </DataTemplate>
        </ListView.ItemTemplate>
    </ListView>
```
3 Initialize the list in the constructor:
``` C#
	public MainPage()
	{
		InitializeComponent();

		ItemsList.ItemsSource = new string[] { "Hello", "World" }; 
	}
```

### Version with bug

Release Candidate 1 (current)

### Last version that worked well

Unknown/Other

### Affected platforms

Android, Windows, I was *not* able test on other platforms

### Affected platform versions

Windows 10.0.17763.0, Android 10

**Comment by kristinx0211** [comment]
@pekspro I verified this issue on vs 17.3.0 Preview 1.0 [32413.517.main].
on windows, menu item can show when the user right clicked. see:
![image](https://user-images.githubusercontent.com/99457000/163674998-874e4054-dea9-4d5f-99cc-ec129adc2096.png)
on android 12, not crash, but menu item not show when pointer were when the user right clicked.
![image](https://user-images.githubusercontent.com/99457000/163675170-9785a932-a859-4e6b-950a-424f0db60729.png)

**Comment by PureWeen** [comment]
I copied the windows portion of this issue to a new [feature request](https://github.com/dotnet/maui/issues/6490). The positioning hasn't changed here from Xamarin.Forms.  I checked if there was an easy way to make the context menu show up under the mouse click but didn't find anything obvious. The popup is using a flyout menu that you attach to an element and the place apis only really allow you to place it around the element not at a specific location (AFAICT)
